### ISSUE TITLE
When authorization is removed the widget keeps loading

### ISSUE THREAD

**Issue by mitogh** [google/site-kit-wp#4542]
## Bug Description

Given a user that is authorized to a module (`Analytics`), after the user is granted the appropriate permissions and the user connects both modules if the user no longer has access to the data of those modules (the user permissions were removed from `Analytics` or `Search Console` the module never terminates the loading process and remains to load when visiting the modules section.

## Steps to reproduce

1. First assign a user to a property in Google Analytics.
2. Then connect your Site Kit Plugin with this assigned user, 
3. Then connect the Google Analytics module
4. Observe the module working as expected.
5. Then remove this user access from the Google Analytics page so this user no longer has access to this report.
6. Then from your browser visiting the Site Kit dashboard remove all the Session Storage values by going in Settings > More Tools > Developer Tools. Click on Application > Session Storage and remove all the values from the site kit
7. Refresh the page and observe the page showing the loading indicator every time you refresh the page.

## Screenshots

**Analytics**
![2021-12-10_14-52](https://user-images.githubusercontent.com/3921289/145639911-76d746e7-a646-4e94-8be3-13735523de13.png)

**Search Console**

![2021-12-10_14-52_1](https://user-images.githubusercontent.com/3921289/145639915-7b9614e5-d841-4c71-82b2-f277cff9017f.png)

## Additional Context

 - PHP Version: 7.3
 - OS: Manjaro
 - Browser: Chrome
 - Plugin Version : 1.47.0
 - Device: Desktop

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
@wpdarren would you please give this a test to see if you're able to reproduce it?

**Comment by wpdarren** [comment]
@aaemnnosttv yes, I was able to reproduce this.  

Rather than seeing the `User does not have sufficient permissions` message, the flashing loading indicator appears on the Analytics module page and the current and unified dashboard. 

<details>
 <summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/73545194/151966466-83c176cc-8369-4777-bea7-02bfc262332f.png)

![image](https://user-images.githubusercontent.com/73545194/151966499-f7ca126c-2cb0-4195-a2fe-ca7edcb15419.png)

</details>

**Comment by aaemnnosttv** [comment]
@eugene-manuilov I just realized the suggested changes also apply to our `hasZeroData` selectors which are very similar in their implementation so I've updated the IB to include these as well and bumped the estimate accordingly – does that sound okay to you?

**Comment by eugene-manuilov** [comment]
Yes, looks good to me, @aaemnnosttv. Thanks for adding it! IB ✔️

**Comment by aaemnnosttv** [comment]
Moving this to the next release due to some unforeseen side effects that have come up as a result of these changes.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- Could reproduce the error on the latest release.
- When switched to develop branch, I could see that the widgets will no longer be stuck on loading state and they load the error state of the widgets.

![image](https://user-images.githubusercontent.com/73545194/165957421-fb7850d4-668b-43fe-ba7a-56a5e64b5c38.png)
